### PR TITLE
Corrections to prevent errors and warnings

### PR DESCRIPTION
**Pull request by sheridancbio**
- OncotreeUtilTestConfig Beans return concrete types (rather than interfaces) to avoid Bean selection problems on some platforms (linux)
- TopBraidRepository uses annonymous type parameter specifier to avoid unchecked type conversion warnings

### CODE REVIEW

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java [hunk 1]
```diff
@@ -147,13 +145,12 @@ private Version setupMockVersion() {
...
+        return expectedTumorTypeMap;
     }
-    
-    
-    private List<MainType> setupExpectedMainTypeList() throws Exception {
+
+    private ArrayList<MainType> setupExpectedMainTypeList() throws Exception {
         Set<MainType> expectedMainTypeSet = new HashSet<>();
-        List<MainType> expectedMainTypeList = new ArrayList<>();
+        ArrayList<MainType> expectedMainTypeList = new ArrayList<>();
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java]: Why is this necessary?

**Comment by sheridancbio** [comment]
Improvement and reversion to generic collection interface types planned after release.
